### PR TITLE
Allow "Compute an expression on every row" to succeed with empty input

### PR DESCRIPTION
**Pull request by wm75**
Currently the tool fails when its input doesn't have at least one line of content, which is a nuisance with more complex WFs and collections.

While working on this I tried to improve the tool in a few other ways, too.

### CODE REVIEW

File: tools/column_maker/column_maker.xml [hunk 1]
```diff
@@ -59,19 +62,19 @@
         <test>
             <param name="cond" value="c3-c2"/>
             <param name="input" value="1.bed"/>
-            <param name="round" value="no"/>
+            <param name="round" value="False"/>
```
Comment (mvdbeek) [inline tools/column_maker/column_maker.xml:65]:
```suggestion
            <param name="round" value="false"/>
```

File: tools/column_maker/column_maker.xml [hunk 2]
```diff
@@ -59,19 +62,19 @@
...
             <output name="out_file1" file="column_maker_out1.interval"/>
         </test>
         <test>
             <param name="cond" value="c4*1"/>
             <param name="input" value="1.interval"/>
-            <param name="round" value="no"/>
+            <param name="round" value="False"/>
```
Comment (mvdbeek) [inline tools/column_maker/column_maker.xml:71]:
```suggestion
            <param name="round" value="false"/>
```

File: tools/column_maker/column_maker.xml [hunk 3]
```diff
@@ -59,19 +62,19 @@
...
             <output name="out_file1" file="column_maker_out2.interval"/>
         </test>
         <test>
             <param name="cond" value="c4*1"/>
             <param name="input" value="1.header.tsv"/>
-            <param name="round" value="no"/>
+            <param name="round" value="False"/>
```
Comment (mvdbeek) [inline tools/column_maker/column_maker.xml:77]:
```suggestion
            <param name="round" value="false"/>
```

File: tools/column_maker/column_maker.xml [hunk 4]
```diff
@@ -81,13 +84,13 @@
         <test>
             <param name="cond" value="c4*1"/>
             <param name="input" value="1.interval"/>
-            <param name="round" value="yes"/>
+            <param name="round" value="True"/>
```
Comment (mvdbeek) [inline tools/column_maker/column_maker.xml:87]:
```suggestion
            <param name="round" value="true"/>
```

File: tools/column_maker/column_maker.xml [hunk 5]
```diff
@@ -81,13 +84,13 @@
...
             <output name="out_file1" file="column_maker_out3.interval"/>
         </test>
         <test>
             <param name="cond" value="float(.0000000000001)"/>
             <param name="input" value="1.bed"/>
-            <param name="round" value="no"/>
+            <param name="round" value="False"/>
```
Comment (mvdbeek) [inline tools/column_maker/column_maker.xml:93]:
```suggestion
            <param name="round" value="false"/>
```

File: tools/column_maker/column_maker.xml [hunk 6]
```diff
@@ -98,8 +101,8 @@
         <test>
             <param name="cond" value="float(.0000000000001)"/>
             <param name="input" value="1.bed"/>
-            <param name="round" value="no"/>
-            <param name="avoid_scientific_notation" value="yes"/>
+            <param name="round" value="False"/>
```
Comment (mvdbeek) [inline tools/column_maker/column_maker.xml:104]:
```suggestion
            <param name="round" value="false"/>
```

File: tools/column_maker/column_maker.xml [hunk 7]
```diff
@@ -98,8 +101,8 @@
...
+            <param name="avoid_scientific_notation" value="True"/>
```
Comment (mvdbeek) [inline tools/column_maker/column_maker.xml:105]:
```suggestion
            <param name="avoid_scientific_notation" value="true"/>
```

**Review by mvdbeek — COMMENTED** [review]
Looks good, only comment is that boolean values should be lowercase.

**Comment by wm75** [comment]
Ah, of course, you're right. This whole wrapper/json/python dance is confusing.

**Review by mvdbeek — APPROVED** [review]

**Comment by nsoranzo** [comment]
I just noticed that `column_maker` was migrated to tools-iuc in https://github.com/galaxyproject/tools-iuc/pull/2469 , not sure why it wasn't removed from here!

@wm75 Would you be interested in merging it again into tools-iuc and remove it from here for good?

**Comment by wm75** [comment]
Yes, just read @jmchilton's https://github.com/galaxyproject/tools-iuc/issues/3494.
I can do this tonight.

**Comment by nsoranzo** [comment]
@wm75 Thanks! I realise the 2 have diverged significantly, so let me know if you'd like help or change your mind!

**Comment by jmchilton** [comment]
Thanks so much!

**Comment by wm75** [comment]
@nsoranzo @jmchilton @mvdbeek addressed by #586.
Turned out the merge is not *that* complicated this way round.
After merging the new PR the tools-devteam version can finally be removed and the new version moved over to tools-iuc (will produce a rather ugly diff over there, but the tools-iuc version was just too far behind).

**Pull request merged**